### ISSUE TITLE
ConfigurationChanges not working

### ISSUE THREAD

**Issue by mbmleone** [xamarin/xamarin-android#2963]
It seems Xamarin.Android is not applying 
`ConfigurationChanges = ConfigChanges.Keyboard | ConfigChanges.KeyboardHidden`
when added to Activity .
This problem was already described here years ago:
https://bugzilla.xamarin.com/show_bug.cgi?id=25888

If you read it you see that it was already stated that `ConfigChanges.Keyboard` doesn't work.
(See comments 3 and 4)
It seems the writer of comment 11 didn't understand this and gave standard answer and still closed the issue.

We have added the mentioned line and it's really not working while `ConfigChanges.ScreenSize | ConfigChanges.Orientation` do work as expected.
With `ConfigChanges.Keyboard | ConfigChanges.KeyboardHidden` the Activity however still restarts which should not be the case.

Can someone please look into this?
Thanks!

**Comment by pjcollins** [comment]
@mbmleone are you using the same test scenario described in 25888 which involves a bluetooth keyboard? I don't have a relevant device with me right now but I can try to take a look next week.

Could you check the content of the `AndroidManifest.xml` file which is output at `obj\Release\81\android` to ensure it contains the correct `android:configChanges` value? (e.g. `android:configChanges="keyboard|keyboardHidden|orientation|screenSize"`).

**Comment by mbmleone** [comment]
@pjcollins thanks for your quick response.
Yes, our scenario is the same as the original.
Today I have created a better example with the files from original poster:
[Android  - SerialNumber.zip](https://github.com/xamarin/xamarin-android/files/3097578/Android.-.SerialNumber.zip)

Here you can see that it still restart even if `ConfigChanges.Keyboard | ConfigChanges.KeyboardHidden` is added.
I have checked AndroidManifest.xml generated, it does contain `keyboard|keyboardHidden` in `android:configChanges` .

Hopefully you can find out more with this information.

**Comment by mbmleone** [comment]
Because the AndroidManifest.xml was correct I did some more research today.
Then I found this information:
[https://stackoverflow.com/questions/25735227/bluetooth-keyboard-will-cause-activity-destroy-and-recreate](url)
Here it was solved by adding `navigation` to `android:configChanges` .
After added `ConfigChanges.Navigation` to `ConfigurationChanges` the problem went away.
It now doesn't restart anymore.
So no Xamarin.Android problem, but a problem in Android itself.
